packages/ui/src/Card.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { FC } from 'react'
+import { Icon, IconProps } from './Icon'
+import { DataTestProp } from '@hazelcast/helpers'
+import cn from 'classnames'
+
+import { IconButton, IconButtonProps } from './IconButton'
+
+import styleConsts from '../styles/constants/export.scss'
+import styles from './Card.module.scss'
+
+type CardHeaderProps =
+  | {
+      title: string
+      icon?: IconProps['icon']
+      iconButtonProps?: never
+    }
+  | {
+      title: string
+      iconButtonProps?: Omit<IconButtonProps, 'size' | 'kind'>
+      icon?: never
+    }
+  | {
+      title?: never
+      iconButtonProps?: never
+      icon?: never
+    }
+
+export type CardProps = {
+  type?: 'primary' | 'secondary' | 'highlighter'
+} & CardHeaderProps &
+  DataTestProp
+
+/**
+ * ### Purpose
+ * Card components define content hierarchy (visual and logical), separating content into "groups".
+ *
+ * ### General Info
+ * - There are three types of Cards - "Primary", "Secondary" and "Highlighter".
+ * - "Secondary" and "Highlighter" Cards should be used only within "Primary" Cards.
+ * - Card can display an icon next to Card heading. This icon can be static or interactive (button).
+ */
+export const Card: FC<CardProps> = ({ type = 'primary', title, icon, iconButtonProps, 'data-test': dataTest, children }) => {
+  let iconElement: JSX.Element | null = null
+
+  if (icon !== undefined) {
+    iconElement = <Icon icon={icon} color={styleConsts.colorPrimary} size={type === 'primary' ? 'normal' : 'small'} ariaHidden />
+  }
+
+  if (iconButtonProps !== undefined) {
+    iconElement = (
+      <IconButton kind="transparent" color={styleConsts.colorPrimary} size={type === 'primary' ? 'normal' : 'small'} {...iconButtonProps} />
+    )
+  }
+
+  return (
+    <div
+      data-test={dataTest ?? 'card-wrapper'}
+      className={cn(styles.wrapper, {
+        [styles.typeSecondary]: type === 'secondary',
+        [styles.typeHighlighter]: type === 'highlighter',
+      })}
+    >
+      {title && (
+        <>
+          {type === 'primary' ? (
+            <h2 data-test="card-heading" className={styles.heading}>
+              {title}
+              {iconElement && <span className={styles.icon}>{iconElement}</span>}
```

Comment:
@jozefhruska Is the icon supposed to be on the right? In figma example, it is on the left: https://www.figma.com/file/8mVm6LTbp2Z0RaWWjTZoft/%F0%9F%90%9DHIVE---Hazelcast-Design-System?node-id=10077%3A2

Reply:
When I started there were only Zeroheight designs. We should probably review designs entirely, I'll ask Pawel.